freqtrade/persistence/models.py

```diff
@@ -577,18 +577,19 @@ def recalc_trade_from_orders(self):
 
         total_amount = 0.0
         total_stake = 0.0
-        for temp_order in self.orders:
-            if (temp_order.ft_is_open or
-                    (temp_order.ft_order_side != 'buy') or
-                    (temp_order.status not in NON_OPEN_EXCHANGE_STATES)):
+        for o in self.orders:
+            if (o.ft_is_open or
+                    (o.ft_order_side != 'buy') or
+                    (o.status not in NON_OPEN_EXCHANGE_STATES)):
                 continue
 
-            tmp_amount = temp_order.amount
-            if temp_order.filled is not None:
-                tmp_amount = temp_order.filled
-            if tmp_amount > 0.0 and temp_order.average is not None:
+            tmp_amount = o.amount
+            tmp_price = o.average or o.price or 0.0
+            if o.filled is not None:
+                tmp_amount = o.filled
+            if tmp_amount > 0.0 and tmp_price is not None and o.status == 'closed':
```

Comment:
i don't think you should check for `'closed'` here explicitly. That will exclude partially filled (but cancelled ) orders - and this filter is already in place in L583.

Reply:
Yes, this would fail with partial fills. My bad, fixed.